lib/src/routing/connection_registry.rs

```diff
@@ -1,350 +1,221 @@
-                                }
-                            };
-                        }
-                        Some(RegistryCommand::Stop) | None => {
-                            debug!("Stopping background updater");
-                            break;
-                        }
+            } else {
+                match self
+                    .fetch_routing_table(db.clone(), imp_user.clone(), bookmarks)
+                    .await
+                {
+                    Ok(new_table) => {
+                        let database_table: DatabaseTable = new_table.into();
+                        let servers = database_table.resolve();
+                        debug!("Routing table for database {db_name} refreshed");
+                        servers
+                    }
+                    Err(e) => {
+                        error!("Failed to refresh routing table for database {db_name}: {e}");
+                        vec![]
                     }
                 }
             }
-
-            debug!("Resetting interval with TTL: {}", ttl);
-            // recreate interval with the new TTL or the derived one in case of a forced update
-            interval = tokio::time::interval(Duration::from_secs(ttl));
-            interval.tick().await;
-        }
-    });
-    tx
-}
-
-impl ConnectionRegistry {
-    /// Retrieve the pool for a specific server and database.
-    pub fn get_pool(&self, server: &BoltServer) -> Option<ConnectionPool> {
-        self.pool_registry
-            .get(server)
-            .map(|pool| pool.value().clone())
-    }
-
-    /// Mark a server as available for a specific database.
-    pub fn mark_unavailable(&self, server: &BoltServer, db: Option<Database>) {
-        let db_name = self.get_db_name(db);
-        if self.databases.contains_key(db_name.as_str()) {
-            if let Some(index) = self
-                .databases
-                .get(db_name.as_str())
-                .and_then(|vec| vec.iter().position(|s| server.has_same_address(s)))
+        } else {
+            match self
+                .fetch_routing_table(db.clone(), imp_user.clone(), bookmarks)
+                .await
             {
-                debug!("Marking server as available: {:?}", server);
-                self.databases
-                    .get_mut(db_name.as_str())
-                    .unwrap()
-                    .remove(index);
-                self.pool_registry.remove(server);
-            } else {
-                debug!("Server not found in the registry: {:?}", server);
+                Ok(new_table) => {
+                    let db = new_table.db.clone();
+                    let database_table: DatabaseTable = new_table.into();
+                    let servers = database_table.resolve();
+                    let db_name = db.map_or(String::new(), |d| d.to_string());
+                    debug!("Routing table for database {db_name} refreshed");
+                    servers
```

Comment:
Moving the log statement up means we can print from new_table directly without clone and to_string:

```suggestion
                    let db_name = new_table.db.as_deref().unwrap_or("");
                    debug!("Routing table for database {db_name} refreshed");
                    let database_table: DatabaseTable = new_table.into();
                    database_table.resolve()
```